jcache/embedded/src/main/java/org/infinispan/jcache/embedded/JCacheListenerAdapter.java

```diff
@@ -37,14 +37,14 @@ public void handleCacheEntryCreatedEvent(CacheEntryCreatedEvent<K, V> e) {
    public void handleCacheEntryModifiedEvent(CacheEntryModifiedEvent<K, V> e) {
       // JCache listeners notified only once, so do it after the event
       if (!e.isPre() && !e.isCreated())
-         notifier.notifyEntryUpdated(jcache, e.getKey(), e.getValue());
+         notifier.notifyEntryUpdated(jcache, e.getKey(), e.getValue(), e.getValue());
    }
 
    @CacheEntryRemoved
    public void handleCacheEntryRemovedEvent(CacheEntryRemovedEvent<K, V> e) {
       // JCache listeners notified only once, so do it after the event
       if (!e.isPre())
-         notifier.notifyEntryRemoved(jcache, e.getKey(), e.getOldValue());
+         notifier.notifyEntryRemoved(jcache, e.getKey(), e.getOldValue(), e.getOldValue());
```

Comment:
Current value should be `null`, shouldn't it?

Reply:
No, it's correct as is. As per the CacheEntryEvent javadoc for getValue():

    * Returns the same value as {@link #getOldValue()} for
    * {@link CacheEntryExpiredListener} and {@link CacheEntryRemovedListener}.